cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines-overwrite-autodetected-variables/meta.json

```diff
@@ -20,6 +20,7 @@
     "scan_environment": "azure-pipelines",
     "is_full_scan": true,
     "is_sca_scan": false,
```

Comment:
would be better to define a proper product type in semgrep_output_v1.atd
with Sca | Code | Secrets | ...
and then keep here instead a list of those products instead of 10 booleans.

Reply:
I agree that this interface would be better if it was typed. I don't think this PR is the right place for that. Maybe we can add types (or ask someone else to) in preparation for the switch to osemgrep?

Reply:
One nice thing about @aryx's proposed type representation is that it would be more stable to changes in these snapshots.

Reply:
agree with typing the interface, I can try to put up a PR for that later, but would rather not block this work on that change. Changing the interface is a whole can of worms, because the app always has to stay backwards-compatible with old versions of the CLI, so we will have to maintain backwards compatibility with the list of booleans approach anyway. I think we should hold off on that change until the interface is strongly typed (I will try to get to that this week).

Reply:
That's exactly why I'd rather make it better now, because otherwise we will have to support this ugly is_code_scan boolean forever.